pkg/api/mlflow/dao/repositories/param.go

```diff
@@ -31,8 +32,50 @@ func NewParamRepository(db *gorm.DB) *ParamRepository {
 
 // CreateBatch creates []models.Param entities in batch.
 func (r ParamRepository) CreateBatch(ctx context.Context, batchSize int, params []models.Param) error {
-	if err := r.db.CreateInBatches(params, batchSize).Error; err != nil {
+	var err error
+	err = r.db.CreateInBatches(params, batchSize).Error
+	if err != nil {
+		// remove duplicate rows and try again
+		params, err1 := r.removeExactMatches(ctx, params)
+		if err1 != nil {
+			return eris.Wrap(err1, "error removing exact matches after: " + err.Error())
+		}
+		err = r.db.CreateInBatches(params, batchSize).Error
+	}
+
+	if err != nil {
 		return eris.Wrap(err, "error creating params in batch")
 	}
 	return nil
 }
+
+// removeExactMatches will return a new slice of params which excludes exact matches
+func (r ParamRepository) removeExactMatches(ctx context.Context, params []models.Param) ([]models.Param, error) {
+	var keys, foundKeys []string
+	paramMap := map[string]models.Param{}
+	paramsToReturn := []models.Param{}
```

Comment:
can we please init variable new the place where we use it?

Reply:
moved